src/lib/components/Gallery.svelte

```diff
@@ -5,13 +5,14 @@
 	import { faBan } from '@fortawesome/free-solid-svg-icons'
 
 	$: noImage = $galleryStore.selectedTag && !$galleryStore.selectedFilteredImages.length
+
+	$: imageHeight = $galleryStore.zoomLevel + 80
 </script>
 
 <div
-	class="gap-3 flex-1 w-full overflow-auto
-		{noImage
-		? 'flex flex-col gap-5 items-center w-full justify-center'
-		: 'grid grid-cols-2 md:grid-cols-4 xl:grid-cols-6'}"
+	class="w-full gap-3
+		{noImage ? 'flex flex-col items-center justify-center gap-5' : 'grid grid-flow-row'}"
+	style="grid-template-columns: repeat(auto-fill, minmax({imageHeight}px, 1fr));"
```

Comment:
@alvinosh 
With this approach, the zooming feature doesn't work smoothly. 
In my test, zoom from 100 to 125, doesn't have any changes to images, which is wrong and bad UX.
Zooming should work as smoothly.